test/Microsoft.ML.Tests/AnomalyDetectionTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.IO;
+using System.Linq;
+using Microsoft.Data.DataView;
+using Microsoft.ML.Data;
+using Microsoft.ML.ImageAnalytics;
+using Microsoft.ML.Model;
+using Microsoft.ML.RunTests;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests
+{
+    public class AnomalyDetectionTests : TestDataPipeBase
+    {
+        public AnomalyDetectionTests(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        /// <summary>
+        /// RandomizedPcaTrainer test 
+        /// </summary>
+        [Fact]
+        public void RandomizedPcaTrainer()
+        {
+            var mlContext = new MLContext(seed: 1, conc: 1);
+            string featureColumn = "NumericFeatures";
+
+            var reader = new TextLoader(Env, new TextLoader.Arguments()
+            {
+                HasHeader = true,
+                Separator = "\t",
+                Columns = new[]
+                {
+                    new TextLoader.Column("Label", DataKind.R4, 0),
+                    new TextLoader.Column(featureColumn, DataKind.R4, new [] { new TextLoader.Range(1, 784) })
+                }
+            });
+
+            var trainData = reader.Read(GetDataPath(TestDatasets.mnistOneClass.trainFilename));
+            var testData = reader.Read(GetDataPath(TestDatasets.mnistOneClass.testFilename));
+
+            var pipeline = ML.AnomalyDetection.Trainers.RandomizedPca(featureColumn);
+
+            var transformer = pipeline.Fit(trainData);
+            var transformedData = transformer.Transform(testData);
+
+            // Evaluate
+            var metrics = ML.AnomalyDetection.Evaluate(transformedData);
+
+            Assert.Equal(0.99, metrics.Auc, 2);
```

Comment:
How do we know that these numbers are correct?

Reply:
i tried out the same dataset in TLC with the same trainer, the numbers are *close*.

in general, in this PR i am only exposing the  trainer / evaluators  as they exist  currently in the codebase.  the PR does not have  any algorithmic changes or  changes in evaluation metrics themselves.

---
In reply to: [255589211](https://github.com/dotnet/machinelearning/pull/2390#discussion_r255589211) [](ancestors = 255589211)

Reply:
Can we mark that this is a baseline test, then, and not a functionality test?

---
In reply to: [255680814](https://github.com/dotnet/machinelearning/pull/2390#discussion_r255680814) [](ancestors = 255680814,255589211)

Reply:
good point. 

as per the classification above, these seem like "baseline" tests, I have increased the precision to 5 places of decimal.

as for test migration from internal repo, it seems we ported over the `PcaAnomalyTest`  already  . that should suffice for correctness.

---
In reply to: [255796608](https://github.com/dotnet/machinelearning/pull/2390#discussion_r255796608) [](ancestors = 255796608,255680814,255589211)